torch/csrc/jit/autodiff.cpp

```diff
@@ -128,6 +128,12 @@ bool isDifferentiable(Node* n) {
   if (differentiable_ops.find(n))
     return true;
 
+  if (n->matches(
+          "aten::dropout(Tensor input, float p, bool train) -> Tensor")) {
+    auto train = n->get<bool>(attr::train).value();
```

Comment:
You never checked that this is a constant. If `train` is a computed value, this will throw. You have to do `n->matches("...", attr::train)` above.

Reply:
also nit: no need to assign to a local just to return